elastic/apm-server
# script: simplify generate_notice.py by using go-licence-detector.

## Motivation/summary

Outsource the heavy lifting to https://github.com/elastic/go-licence-detector, which is now used by Beats. We mostly use the beats-provided templates, but there is a small amount of customisation:
 - change product name from "Elastic Beats" to "Elastic APM Server"
 - add Beats itself to the list of licence overrides

Also, unlike Beats we do not use `go list -m all`. Instead, we (continue to) list only the packages which are linked into the final binary, and include their modules. This excludes numerous modules which are included only for tests.

## How to test these changes

- Run `make notice`, and check NOTICE.txt looks sensible
- Run `python3 scripts/generate_notice.py --csv=dependencies.csv`, and check `dependencies.csv` looks sensible (compare to previous version)

## Related issues

Closes #3516
